ejb/WFLY-13354-galleon-for-ejb.adoc

```diff
@@ -0,0 +1,110 @@
+= Galleon layers for the EJB subsystem
+:author:            Brian Stansberry; Yeray Borges; Cheng Fang
+:email:             cfang@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+The goal of this feature is to supply a list of https://docs.wildfly.org/galleon/#_layers[Galleon layers]
+for the EJB subsystem to support common predefined EJB feature sets.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFLY-13354[WFLY-13354 Add Galleon layers for the EJB subsystem]
+* https://issues.redhat.com/browse/EAP7-1444[EAP7-1444 Galleon Layers in EAP XP 2]
+
+=== Related Issues
+* https://issues.redhat.com/browse/WFLY-13433[Improve capability support in EJB3 subsystem]
+* https://issues.redhat.com/browse/WFLY-13356[Add a Galleon layer for Webservices]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:fburzigo@redhat.com[Fabio Burzigotti]
+
+=== Testing By
+
+* [X] Engineering
+
+* [ ] QE
+
+=== Affected Projects or Components
+
+WildFly
+
+=== Other Interested Projects
+
+== Requirements
+
+These analysis and requirements are based on https://github.com/bstansberry/wildfly/commits/ejb-layers2[Brian's proposal].
+
+If a user is trying to slim their server I think of four key things they want to do in descending order of importance:
+
+1. Eliminate unnecessary open ports.
+
+2. Eliminate unnecessary exposure.
+
+3. Eliminate jars, both to save filesystem footprint and to reduce any theoretical attack surface.
+
+4. Reduce configuration clutter and unnecessary services.
+
+Based on the above factors, the following layers are to be supported:
+
+1. ejb3-lite. This is like what we provide in standalone.xml, but without the MDB instance pool or the remote connector resource.
+(A webservices layer would depend on this.)
+
+2. ejb3. Builds (i.e. depends) on ejb3-lite and adds the discovery subsystem, the MDB pool and the remote connector resource.
+This looks what we provide in standalone-full.xml, except no IIOP integration.
+
+3. ejb3-iiop. Builds (i.e. depends) on *ejb3-lite* and adds the iiop subsystem plus the integration resource in the EJB3 subsystem.
```

Comment:
I had to pause the work but the IIOP side may need this issue resolved as the legacy security subsystem will not be available as a layer https://issues.redhat.com/browse/WFLY-13679

Reply:
@darranl shall we mention WFLY-13679 as a blocker for the current RFE?

Reply:
For this piece I will try and look again when I am back, but we do need to make sure any supported layers don't inadvertently pull in the legacy security subsystem - I would say if they do we should raise that as a Blocker until we can triage.

Having said that some layers will still pull in the PicketBox modules but we can slowly work to clean those up.

Reply:
@tadamski can you take a look at the iiop-openjdk's dependency on picketbox (https://issues.redhat.com/browse/WFLY-13679)? Is its dependency on legacy security really required?

Reply:
@chengfang, @darranl, @tadamski does this conversation prevents the AD from being approved? do you think we can approve the AD and move this conversation somewhere else?

Reply:
Let's investigate it ([WFLY-13679](https://issues.redhat.com/browse/WFLY-13679) Make legacy security optional for "org.wildfly.iiop-openjdk") separately. I think it should not be blocking this proposal.

Reply:
We'll drop ejb-iiop and iiop-openjdk layers from this feature. Once WFLY-13679 is resolved, we can then add them back.